src/components/PostsSection.js

```diff
@@ -1,21 +1,89 @@
-import React from "react";
-import Section from "./Section";
-import Container from "react-bootstrap/Container";
-import SectionHeader from "./SectionHeader";
-import Alert from "react-bootstrap/Alert";
-import Row from "react-bootstrap/Row";
-import Col from "react-bootstrap/Col";
-import Card from "react-bootstrap/Card";
-import UserPreferences from "./UserPreferences";
+import React, { useState, useRef } from "react";
+import { Container, Form, Spinner, Button, Row, Col } from "react-bootstrap";
 import { Link, useRouter } from "./../util/router.js";
 import { useAuth } from "./../util/auth.js";
+import { useForm } from "react-hook-form";
+import FormField from "./FormField";
+import Section from "./Section";
+import SectionHeader from "./SectionHeader";
+import { ageGroups, eligibilityGroups, provinces, error, selectAll } from "./formConstants";
 
 function PostsSection(props) {
   const auth = useAuth();
   const router = useRouter();
-  
+  const { handleSubmit, register, errors } = useForm();
+
+  const [numPostals, setNumPostals] = useState(1);
+  const [pending, setPending] = useState(false);
+  const [groupError, setGroupError] = useState(false); 
+  const [areaError, setAreaError] = useState(false);
+
   let accountConfigured = auth.user.phone && auth.user.province && auth.user.postalcode;
 
+  const onSubmit = (data) => {
+    const allPostalCodes = document.querySelectorAll(".postalCodesInput");
+    const province = document.querySelector(".province");
+
+    if (document.querySelectorAll('input[type="checkbox"]:checked').length === 0) {
+      setGroupError(true);
+      const allCheckBoxes = document.querySelectorAll('input[type="checkbox"]');
+      for (let i = 0; i < allCheckBoxes.length; i++) {
+        allCheckBoxes[i].addEventListener("click", function() {
+          setGroupError(!this.checked)
+        });
+      }
+    }
+
+    data.postal = [];
+    allPostalCodes.forEach((postal) => {
+      if (!data.postal.includes(postal)) {
```

Comment:
if they added an empty postal code we should filter that out (if (postal && !data.postal.inclu...)